libglusterfs/src/common-utils.c

```diff
@@ -1767,35 +1767,25 @@ gf_is_ip_in_net(const char *network, const char *ip_str)
         goto out;
     }
 
-    net_str = strdupa(network);
-    slash = strchr(net_str, '/');
-    if (!slash)
+    size = inet_net_pton(family, network, net_buff, sizeof(net_buff));
+    if (size < 0) {
+        GF_LOG_E("common-utils", LG_MSG_INET_NET_PTON_FAILED(errno));
         goto out;
-    *slash = '\0';
-
-    subnet = slash + 1;
-    net_ip = net_str;
-
-    /* Convert IP address to a long */
-    ret = inet_pton(family, ip_str, &ip_buf);
-    if (ret < 0)
-        gf_smsg("common-utils", GF_LOG_ERROR, errno, LG_MSG_INET_PTON_FAILED,
-                NULL);
-
-    /* Convert network IP address to a long */
-    ret = inet_pton(family, net_ip, &net_ip_buf);
+    }
+    ret = inet_net_pton(family, ip_str, ip_buff, sizeof(ip_buff));
```

Comment:
This one must be `inet_pton()`. It's not a network address.

Reply:
ack